app/src/main/java/com/hedvig/app/feature/profile/ui/tab/ProfileFragment.kt

```diff
@@ -151,16 +89,72 @@ class ProfileFragment : Fragment(R.layout.profile_fragment) {
             .launchIn(lifecycleScope)
     }
 
-    private fun getPriceCaption(data: ProfileQuery.Data, monetaryMonthlyNet: String): String {
-        return marketManager.market?.getPriceCaption(data)?.let {
-            getString(it, monetaryMonthlyNet)
-        } ?: ""
+    private fun buildProfileModelList(profileUiState: ProfileUiState): List<ProfileModel> {
+        return buildList {
+            add(ProfileModel.Title)
+            add(
+                ProfileModel.Row(
+                    title = getString(R.string.PROFILE_MY_INFO_ROW_TITLE),
+                    caption = profileUiState.contactInfoName,
+                    icon = R.drawable.ic_contact_information,
+                    onClick = {
+                        startActivity(Intent(requireContext(), MyInfoActivity::class.java))
+                    }
+                )
+            )
+            when (val charityState = profileUiState.charityState) {
+                CharityState.DontShow -> {}
+                CharityState.NoneSelected -> add(buildCharityRowItem())
+                is CharityState.Selected -> add(buildCharityRowItem(charityState.charityName))
+            }
```

Comment:
On this build function this is the change where the charity isn't shown when not needed when the mapper function determines that the fflag is off.